src/GridCell/GridCell.js

```diff
@@ -21,13 +21,65 @@ export const GridCell = ({ className, children, ref, ...props }) => {
     Element = props.element;
   }
 
-  // props.colSpan
-  if (props.colSpan && gridColSpan.includes(props.colSpan)) {
-    if (props.colSpan === 'auto') {
-      levers.colSpan = `su-col-auto`;
+  // props.xs to props.xxl controls column span of the grid cell
+  // props.xs
+  if (props.xs && gridColSpan.includes(props.xs)) {
+    if (props.xs === 'auto') {
+      levers.xs = `su-col-auto`;
     }
     else {
-      levers.colSpan = `su-col-span-${props.colSpan}`;
+      levers.xs = `su-col-span-${props.xs}`;
+    }
+  }
+
+
+  // props.sm
+  if (props.sm && gridColSpan.includes(props.sm)) {
+    if (props.sm === 'auto') {
+      levers.sm = `sm:su-col-auto`;
+    }
+    else {
+      levers.sm = `sm:su-col-span-${props.sm}`;
+    }
+  }
+
+  // props.md
+  if (props.md && gridColSpan.includes(props.md)) {
+    if (props.md === 'auto') {
+      levers.md = `md:su-col-auto`;
+    }
+    else {
+      levers.md = `md:su-col-span-${props.md}`;
+    }
+  }
+
+  // props.lg
+  if (props.lg && gridColSpan.includes(props.lg)) {
+    if (props.lg === 'auto') {
+      levers.lg = `lg:su-col-auto`;
+    }
+    else {
+      levers.lg = `lg:su-col-span-${props.lg}`;
+    }
+  }
+
+  // props.xl
+  if (props.xl && gridColSpan.includes(props.xl)) {
+    if (props.xl === 'auto') {
+      levers.xl = `xl:su-col-auto`;
+    }
+    else {
+      levers.xl = `xl:su-col-span-${props.xl}`;
+    }
+  }
+
+  // props.xxl
+  if (props.xxl && gridColSpan.includes(props.xxl)) {
+    if (props.xxl === 'auto') {
+      levers.xxl = `2xl:su-col-auto`;
+    }
+    else {
+      levers.xxl = `2xl:su-col-span-${props.xxl}`;
     }
   }
```

Comment:
This whole thing could probably be re-written in a loop to avoid code duplication.

eg:
```
const bps = ['xs', 'sm', 'md', 'xl', '2xl'];
bps.forEach((bp) => {
  if (props[bp] && gridColSpan.includes(props[bp]) {
	if(props[bp] === "auto") {
	  levers[bp] = bp + ':su-col-auto';
	}
	else {
	  levers[bp] = bp + ':su-col-span-' + props[bp];
	}
  }
}
```

Reply:
That's cleaner :)

Reply:
Do you happen to know what's the proper syntax for doing this for destructured props? E.g., in Grid component, the props xs, sm etc are all destructured so I tried the below and a few other permutations but couldn't get it to work (couldn't get `bp` to dynamically reference the destructured data with the same name as the breakpoint instead of a string) and my google fu was failing me 😬 

  ```
bps.forEach((bp) => {
      if ([bp] && gridNumCols.includes([bp])) {
        levers[bp] = `su-grid-cols-${[bp]}`;
      }
    }
  );
```

Reply:
I tried this and it seems to work. Is this ok?

```
let bps = {xs: xs, sm: sm, md: md, lg: lg, xl: xl, xxl: xxl};

  Object.entries(bps).forEach(([key, value]) => {
    if (value && gridNumCols.includes(value)) {
      if (key === 'xs') {
        levers[key] = `su-grid-cols-${value}`;
      }
      else if (key === 'xxl') {
        levers[key] = `2xl:su-grid-cols-${value}`;
      }
      else {
        levers[key] = `${key}:su-grid-cols-${value}`;
      }
    }
  });
```

Reply:
Yeah I went with what worked at the time - I got the syntax wrong somewhere when I tried other approaches. Will ask you later when you have time.